libs/api_common/windows_platform_common.cpp

```diff
@@ -316,7 +316,7 @@ get_ebpf_program_type(bpf_prog_type_t bpf_program_type)
 }
 
 _Ret_maybenull_ const ebpf_attach_type_t*
-get_ebpf_attach_type(bpf_attach_type_t bpf_attach_type) noexcept
+ebpf_get_ebpf_attach_type(bpf_attach_type_t bpf_attach_type) noexcept
```

Comment:
I think this function is subject to a use after free: the returned ebpf_attach_type* is directly returned from _windows_section_definitions. That vector does get cleared when the DLL unloads. So something like:

1. Load DLL
2. Lookup an attach type
3. Unload DLL
4. Use attach type

would cause the use after free. Should I change this function to take an `ebpf_attach_type_t*` as an argument and copy into that?

Reply:
@saxena-anurag what do you think?

Reply:
Until now this was an internal API, so I think there was no issue, But when this is exported, yes in theory it may cause UAF, and we can probably change the signature.

But we may not be hit frequently in real world (if app unloads DLL in step 3, that means it cannot call any eBPF APIs anymore, so it would typically not be using the attach type anymore). Also, this API does not mention that caller needs to free this memory, which means the caller does not own the lifetime of the memory.

But still we can change the signature to make it safer.